pkg/apis/maistra/v2/gateways.go

```diff
@@ -112,9 +115,6 @@ type ClusterIngressGatewayConfig struct {
 	// MeshExpansionPorts define the port set used with multi-cluster/mesh expansion
 	// +optional
 	MeshExpansionPorts []corev1.ServicePort `json:"meshExpansionPorts,omitempty"`
-	// RouteConfig specifies whether to create an OpenShift Route for istio-ingressgateway deployment
-	// +optional
-	RouteConfig *Enablement `json:"routeConfig,omitempty"`
```

Comment:
This removal should trigger removing `spec.gateways.ingress.routeConfig` from the CRD as well, so it seems that something went wrong and `make gen` ignores this change. I don't know why, but from the commit history, I see that you added `spec.gateways.additionalIngress.routeConfig` manually, instead of just modifying the Go struct and running `make gen`. I would try to remove the first commit and then run `make gen`.

Reply:
I did update the go struct and run `make gen` to update the CRDs, but somehow I failed to add `gateways.go` to the first commit.

`spec.gateways.ingress.routeConfig` remains in the CRDs because 'ClusterIngressGatewayConfig' contains (inherits?) the fields from `IngressGatewayConfig`, which is where `RouteConfig` was moved to.

Not sure why the gencheck job is failing... Thought the rebase would fix that. I'll try running `make gen` again.

Reply:
@eoinfennessy if you don't see any changes after running `make gen`, it's probably because you checked out the istio-operator repo somewhere other than under `GOPATH`. I believe `make gen` writes to `$GOPATH/src/github.com/maistra/istio-operator` instead of the current directory. We have to fix this.

Reply:
> spec.gateways.ingress.routeConfig remains in the CRDs because 'ClusterIngressGatewayConfig' contains (inherits?) the fields from IngressGatewayConfig, which is where RouteConfig was moved to.

Ah, right...

Reply:
Thanks for your help @luksa @jewertow. `gencheck` is now succeeding. I'll squash the commits now.